test/jdk/tools/sincechecker/SinceChecker.java

```diff
@@ -0,0 +1,769 @@
+
+note: The `<unique-Element-ID>` for methods looks like `method: <erased-return-descriptor> <binary-name-of-enclosing-class>.<method-name>(<ParameterDescriptor>)`.
+it is somewhat inspired from the VM Method Descriptors. But we use the erased return so that methods that were later generified remain the same.
+*/
+
+
+public class SinceChecker {
+    private final Map<String, Set<String>> LEGACY_PREVIEW_METHODS = new HashMap<>();
+    private final Map<String, IntroducedIn> classDictionary = new HashMap<>();
+    private final JavaCompiler tool;
+    private final List<String> wrongTagsList = new ArrayList<>();
+
+    public static void main(String[] args) throws Exception {
+        if (args.length == 0) {
+            throw new SkippedException("Test module not specified");
+        }
+        SinceChecker sinceCheckerTestHelper = new SinceChecker();
+        sinceCheckerTestHelper.testThisModule(args[0]);
+    }
+
+    private SinceChecker() throws IOException {
+        tool = ToolProvider.getSystemJavaCompiler();
+        for (int i = 9; i <= Runtime.version().feature(); i++) {
+            JavacTask ct = (JavacTask) tool.getTask(null, null, null,
+                    List.of("--release", String.valueOf(i)),
+                    null,
+                    Collections.singletonList(SimpleJavaFileObject.forSource(URI.create("myfo:/Test.java"), "")));
+            ct.analyze();
+
+            String version = String.valueOf(i);
+            Elements elements = ct.getElements();
+            elements.getModuleElement("java.base"); // forces module graph to be instantiated
+            elements.getAllModuleElements().forEach(me ->
+                    processModuleRecord(me, version, ct));
+        }
+    }
+
+    private void processModuleRecord(ModuleElement moduleElement, String releaseVersion, JavacTask ct) {
+        for (ModuleElement.ExportsDirective ed : ElementFilter.exportsIn(moduleElement.getDirectives())) {
+            persistElement(moduleElement, moduleElement, ct.getTypes(), releaseVersion);
+            if (ed.getTargetModules() == null) {
+                analyzePackageRecord(ed.getPackage(), releaseVersion, ct);
+            }
+        }
+    }
+
+    private void analyzePackageRecord(PackageElement pe, String releaseVersion, JavacTask ct) {
+        persistElement(pe, pe, ct.getTypes(), releaseVersion);
+        List<TypeElement> typeElements = ElementFilter.typesIn(pe.getEnclosedElements());
+        for (TypeElement te : typeElements) {
+            analyzeClassRecord(te, releaseVersion, ct.getTypes(), ct.getElements());
+        }
+    }
+
+    private void analyzeClassRecord(TypeElement te, String version, Types types, Elements elements) {
+        Set<Modifier> classModifiers = te.getModifiers();
+        if (!(classModifiers.contains(Modifier.PUBLIC) || classModifiers.contains(Modifier.PROTECTED))) {
```

Comment:
Insert comment:
```
/// JDK documentation only contains public and protected declarations
```
which is 99% true. (The serialization page can contain `private` declarations...)